src/secondary.jl

```diff
@@ -0,0 +1,166 @@
+export run_dssp, run_stride, ss_code!, ss_code, helixselector, sheetselector, coilselector
+
+dssp_executable = `$(DSSP_jll.mkdssp()) --mmcif-dictionary $(DSSP_jll.mmcif_pdbx_dic)`
+const dssp_init_line = "  #  RESIDUE AA STRUCTURE BP1 BP2  ACC     N-H-->O    O-->H-N    N-H-->O    O-->H-N    TCO  KAPPA ALPHA  PHI   PSI    X-CA   Y-CA   Z-CA"
+
+stride_executable = STRIDE_jll.stride_exe()
+
+# Write a PDB file with header and the ATOM records for DSSP/STRIDE to read
+function writesubpdb(filepath::AbstractString, el::ProteinStructure)
+    open(filepath, "w") do io
+        print(
+            io,
+            "HEADER    XXXXXXXXXXXXXXXXXXXXXX                  25-DEC-23   XXXX              ",
+        )
+        for at in collectatoms(el)
+            checkchainerror(at)
+            println(io, pdbline(at))
+        end
+    end
+end
+
+function run_dssp(curr_model::Model, mod_n::Int)
+    struc = ProteinStructure()
+    struc.models[mod_n] = curr_model
+    fixlists!(struc)
+
+    # Write the structure to a temporary PDB file
+    tmp_pdb_path = tempname() * ".pdb"
```

Comment:
Could this be a mmCIF file instead? I can't recall what DSSP/STRIDE can take in. The advantage of mmCIF is that it works for larger structures that break the PDB format.

Reply:
- DSSP supports mmCIF file, STRIDE only supports PDB format.
- DSSP requires certain header information for mmCIF file, unfortunately the output of `writemmcif` will not work. I opened an issue on https://github.com/PDB-REDO/dssp/issues/68.
- Some tools like [maxit](https://sw-tools.rcsb.org/apps/MAXIT/index.html) from rcsb can handle the large structure. They just set the atom serial number to 99999. In this way, we can still get valid ss results from DSSP.